src/tasks/memory.py

```diff
@@ -68,3 +77,463 @@ def _run(self, exp_win, ctl_win):
+        # self.direction_keys = {"left": "a", "right": "d", "up": "w", "down": "s"}  # laptop
+        # self.direction_values = {"a": DIRECTIONS["left"],
+        #                         "d": DIRECTIONS["right"],
+        #                         "w": DIRECTIONS["up"],
+        #                         "s": DIRECTIONS["down"]}  # laptop
+        # self.confidence_keys = {'yes': 'k', 'no': 'l'}  # laptop
+        # self.rating_pygame = {"left": key.A, "right": key.D}  # laptop
+        self.confidence_keys = {'yes': 'a', 'no': 'b'}  # mri
+        self.direction_keys = {"left": "l", "right": "r", "up": "u", "down": "d"}   # mri
+        self.direction_values = {"l": DIRECTIONS["left"],
+                                 "r": DIRECTIONS["right"],
+                                 "u": DIRECTIONS["up"],
+                                 "d": DIRECTIONS["down"]}  # mri
+        self.rating_pygame = {"left": key.L, "right": key.R}  # mri
+
+    def _instructions(self, exp_win, ctl_win):
+        screen_text = visual.TextStim(
+            exp_win,
+            text=self.instruction,
+            alignText="center",
+            color="white",
+            wrapWidth=config.WRAP_WIDTH,
+        )
+        instruction_duration = 10
+        for _ in range(config.FRAME_RATE * instruction_duration):
+            screen_text.draw(exp_win)
+            if ctl_win:
+                screen_text.draw(ctl_win)
+            yield True
+
+        message = (
+            f"In this session, "
+            f"you can earn up to {self.total_possible_points} points.")
+        duration = 3
+        screen_text.text = message
+        for _ in range(config.FRAME_RATE * duration):
+            screen_text.draw(exp_win)
+            if ctl_win:
+                screen_text.draw(exp_win)
+            yield True
+
+    def _setup(self, exp_win):
+        # make pares of rectangels and text stim
+        self.grid = [(
+            visual.TextStim(
+                exp_win, text="", pos=(-0.8 + x * 0.25, 0.7 - y * 0.25),
+                alignText="center", color="white"
+            ),
+            visual.Rect(
+                exp_win, size=(0.2, 0.2), pos=(-0.8 + x * 0.25, 0.7 - y * 0.25),
+                fillColor=None,
+            )
+            )
+            for y, x in itertools.product(range(self.grid_size[1]), range(self.grid_size[0]))
+        ]  # fill the grid from top left corner, left to right first
+        recall_instruction = (f"Use the arrow keys to move the selected box."
+                              "\nI am sure: press "
+                              f"\"{self.confidence_keys['yes']}\" "
+                              "to move on.\nI am not sure: press "
+                              f"\"{self.confidence_keys['no']}\""
+                              " to move on.")
+        self.answer_instruction = visual.TextStim(
+            exp_win, text=recall_instruction,
+            pos=(-0.8 + 2 * 0.25, 0.7 - 5 * 0.25),
```

Comment:
to adapt to above changes, less text wrapping
```suggestion
            pos=(0, (self.grid[-1][0].pos[1]-1)/2),
            wrapWidth=1.9,
```